llm_demo/app.py

```diff
@@ -144,10 +150,65 @@ async def chat_handler(request: Request, prompt: str = Body(embed=True)):
     # Add user message to chat history
     request.session["history"].append({"type": "human", "data": {"content": prompt}})
     orchestrator = request.app.state.orchestrator
-    output = await orchestrator.user_session_invoke(request.session["uuid"], prompt)
+    response = await orchestrator.user_session_invoke(request.session["uuid"], prompt)
+    output = response.get("output")
+    confirmation = response.get("confirmation")
     # Return assistant response
-    request.session["history"].append({"type": "ai", "data": {"content": output}})
-    return markdown(output)
+    if confirmation:
+        return json.dumps({"type": "confirmation", "content": confirmation})
+    else:
+        request.session["history"].append({"type": "ai", "data": {"content": output}})
+        return json.dumps({"type": "message", "content": markdown(output)})
+
+
+@routes.post("/book/flight", response_class=PlainTextResponse)
+async def book_flight(request: Request, params: str = Body(embed=True)):
+    """Handler for LangChain chat requests"""
+    # Retrieve the params for the booking
+    if not params:
+        raise HTTPException(status_code=400, detail="Error: No booking params")
+    if "uuid" not in request.session:
+        raise HTTPException(
+            status_code=400, detail="Error: Invoke index handler before start chatting"
+        )
+    orchestrator = request.app.state.orchestrator
+    response = await insert_ticket(orchestrator, params)
+    # Note in the history, that the ticket has been successfully booked
+    request.session["history"].append(
+        {"type": "ai", "data": {"content": "I have booked your ticket."}}
+    )
+    return response
+
+
+@routes.post("/book/flight/decline", response_class=PlainTextResponse)
+async def decline_flight(request: Request):
+    """Handler for LangChain chat requests"""
+    # Note in the history, that the ticket was not booked
+    # This is helpful in case of reloads so there doesn't seem to be a break in communication.
+    request.session["history"].append(
+        {"type": "ai", "data": {"content": "Please confirm if you would like to book."}}
+    )
+    request.session["history"].append(
+        {"type": "human", "data": {"content": "I changed my mind."}}
+    )
+    return None
+
+
+async def insert_ticket(orchestrator: BaseOrchestrator, params: str):
```

Comment:
Is it better to move this to the orchestrator layer? Since insert_ticket is specific towards the user sesion itself, and also to keep things more consistent

e.g.
in app/py: response = await orchestrator.user_session_insert_ticket()
in BaseOrchestrator:
async user_session_insert_ticket():
    raise NotImplementedError("Subclass should implement this!")

in LangchainOrchestrator:
async user_session_insert_ticket(uuid, param):
    user_sesison = get_user_session(uuid)
    response = user_session.insert_ticket(param)

in UserAgent:
def insert_ticket():
    # call the tool to insert ticket

in tool: (although this won't be a part of the Langchain Tool, it will still be nice to keep it consistent. We just dont add this tool within the `initialize_tool` function)
def insert_ticket()

Reply:
Done! Thank you!